core/src/test/scala/org/apache/spark/scheduler/DAGSchedulerSuite.scala

```diff
@@ -1606,6 +1574,25 @@ class DAGSchedulerSuite
     assertDataStructuresEmpty()
   }
 
+  test("accumulators are updated on exception failures") {
```

Comment:
Clarification question: we're not trying to change behavior for user-defined accumulators, right? If I have a user-defined accumulator and my task fails, then the executor itself will avoid updates to the user accumulators by simply not sending updates for them? I guess this test is just checking that _if_ a failed task sends accumulator updates, then those updates are assumed to be ones that should be counted and are properly added? Might be good to add a one-sentence comment here explaining this if you also find this potentially confusing.